wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -791,6 +823,20 @@ protected void failoverWriter() throws SQLException {
     }
   }
 
+  private void updateHostAvailability(Map<String, HostAvailability> hostAvailabilityMap) {
+    if (hostAvailabilityMap != null && !hostAvailabilityMap.isEmpty()) {
```

Comment:
I feel that it'd be easier to iterate through hostAvailabilityMap and apply availability status to hosts from pluginService.

Reply:
Nested loops :(

Reply:
I think that would be slower because we wouldn't get the advantage of being able to check the map for matching hosts in O(1) time and the hosts are in a list instead of a map. If we say `h` is the number of hosts, `a` is the number of aliases each host has, and `m` is the number of entries in the map, the current code executes in O(h*a) time, but iterating through the map would execute in O(h*a*m) time.